source/gems/dry-validation/comparison-with-activemodel.html.md

```diff
@@ -0,0 +1,606 @@
+
+When using dry-validation, you declare your validation in a separate schema class using predicates to build up rules.
+
+A predicate is a simple stateless method which receives some input and returns either `true` or `false`.
+
+A simple schema can look like this:
+
+```ruby
+schema = Dry::Validation.Schema do
+  required(:email).filled
+  required(:name).filled
+end
+```
+
+## 2. Validation Helpers
+
+### 2.1 acceptance
+
+In ActiveModel validations this helper is used to validate that a checkbox on the user interface was checked when a form was submitted. This is typically used when the user needs to agree to your application's terms of service, confirm reading some text, or any similar concept.
+
+In its simplest form:
+
+**ActiveModel Validation**
+
+```ruby
+validates :attr, acceptance: true
+```
+
+**dry-validation**
+
+```ruby
+required(:attr) { bool? & true? }
+```
+
+When using the `:accepts` option:
+
+**ActiveModel Validation**
+
+```ruby
+validates :attr, acceptance: { accept: 'yes' }
+```
+
+**dry-validation**
+
+```ruby
+required(:attr).filled(eql?: 'yes')
+```
+
+> Note: ActiveModel automatically creates a virtual acceptance attribute for you. If you are using Protected Parameters you will need to add this attribute yourself.
+
+### 2.2 validates_associated
+
+You will need to create a custom predicate to achieve this.
+
+TODO expand
+
+### 2.3 confirmation
+
+This helper is used when you have two text fields that should receive exactly the same content. Common use cases include email addresses and passwords.
+
+**ActiveModel Validation**
+
+```ruby
+validates :attr, confirmation: true
+```
+
+**dry-validation**
+
+```ruby
+required(:attr).confirmation
+```
+
+> Note: ActiveModel automatically creates a virtual confirmation attribute for you whose name is the name of the field that has to be confirmed with "_confirmation" appended. If you are using Protected Parameters you will need to add this attribute yourself.
+
+### 2.4 exclusion
+
+This helper validates that the attributes' values are not included in a given enumerable object.
+
+**ActiveModel Validation**
+
+```ruby
+validates :attr, exclusion: { in: enumerable_object }
+```
+
+**dry-validation**
+
+```ruby
+required(:attr).filled(exclusion?: enumerable_object)
+```
+
+> Note: As per ActiveModel docs, `:within` option is an alias of `:in`
+
+### 2.5 format
+
+This helper validates the attributes' values by testing whether they match or doesn't match a given regular expression.
+
+**ActiveModel Validation**
+
+```ruby
+validates :attr, format: { with: regex }
+```
+
+**dry-validation**
+
+```ruby
+required(:attr).required(format?: regex)
+```
+
+####Doesn't Match
+
+**ActiveModel Validation**
+
+```ruby
+validates :attr, format: { without: regex }
+```
+
+**dry-validation**
+
+```ruby
+required(:attr) { filled? & !format?(regex) }
```

Comment:
This should be `filled? & format?(regex).not` - there's no way to use the standard negation operator (yet). I'll add it in next release